validator_client/src/attestation_producer/mod.rs

```diff
@@ -143,21 +142,8 @@ impl<'a, B: BeaconNodeAttestation, S: Signer, E: EthSpec> AttestationProducer<'a
     }
 
     /// Returns `true` if signing an attestation is safe (non-slashable).
-    ///
-    /// !!! UNSAFE !!!
-    ///
-    /// Important: this function is presently stubbed-out. It provides ZERO SAFETY.
-    fn safe_to_produce(&self, _attestation: &AttestationData) -> bool {
-        //TODO: Implement slash protection
-        true
-    }
-
-    /// Record that an attestation was produced so that slashable votes may not be made in the future.
-    ///
-    /// !!! UNSAFE !!!
-    ///
-    /// Important: this function is presently stubbed-out. It provides ZERO SAFETY.
-    fn store_produce(&mut self, _attestation: &AttestationData) {
-        // TODO: Implement slash protection
+    fn safe_to_produce(&self, attestation: &AttestationData) -> bool {
```

Comment:
Having it return a bool here means we can't have it return the extra info (error info or validity reason). As the code was already written I didn't change it, but I would argue returning an enum and using it upstream would be better.